src/main/java/org/folio/innreach/domain/service/impl/RecordContributionServiceImpl.java

```diff
@@ -109,9 +110,19 @@ private void logItemTransformationError(Item item, Exception e) {
   }
 
   private InnReachResponse contributeBib(UUID centralServerId, String bibId, BibInfo bib) {
-    var response = irContributionService.contributeBib(centralServerId, bibId, bib);
-    Assert.isTrue(response.isOk(), "Unexpected contribution response: " + response);
-    return response;
+    try {
+      var response = irContributionService.contributeBib(centralServerId, bibId, bib);
+      if (!response.getErrors().isEmpty()) {
+        var error = response.getErrors().get(0).getReason();
+        if (error.contains("Contribution to d2irm is currently suspended")) {
+          throw new ServiceSuspendedException("Contribution to d2irm is currently suspended");
+        }
+      }
+      Assert.isTrue(response.isOk(), "Unexpected contribution response: " + response);
+      return response;
+    } catch (ServiceSuspendedException ex) {
+      throw new ServiceSuspendedException(ex.getMessage());
```

Comment:
So we throw an exception, catch the same exception, and re-throw it. What would be the purpose of doing that here? Throwing it once is all you need. There's no need for `try catch` here or in the other methods that do this.